engine/src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -890,4 +880,14 @@ class SlickDataAccess(databaseConfig: Config) extends DataAccess {
 
     runTransaction(action)
   }
+
+  private def materializeWorkflowDescriptor(id: WorkflowId, workflowSourceFiles: WorkflowSourceFiles)(implicit ec: ExecutionContext): Future[WorkflowDescriptor] = {
+    implicit val actorSystem = CromwellServer.actorSystem
+    val materializedWfDescriptorFuture =
+      PromiseActor.askNoTimeout(MaterializeWorkflowDescriptorActor.MaterializeWorkflow(id, workflowSourceFiles), CromwellServer.workflowDescriptorMaterializeActor).mapTo[MaterializationResult]
```

Comment:
**TL;DR We should probably pass in the `actorSystem` and this `actor` as parameters.**

Long-reaching out to singleton `CromwellServer` from inside the database touches on longer standing issue.

NOTE: If he sees this, Jeff may be able to correct me, as I believe he's also spent time in this area.

There are two entrance points to run cromwell, either `Main.run()` or `Main. runServer()`. The way it's currently written, `Main.run()` creates its own `actorSystem`. It doesn't touch the `CromwellServer` singleton object, thus (supposedly) not static invoking the server's separate actor system.

Outside of `Main`, our test specs also repeatedly create their own specialty actor systems as well.

IMHO, the caller to this method should be allowed to pass in whatever actor system it likes.

Thus I vote for plumbing up the call tree, likely all the way to the super class `DataAccess`, method signature that also take in, along with the existing execution context, the system + actor. That way tests can pass in their test actor system, and the workflow manager actors can pass in whatever system they're currently running on.